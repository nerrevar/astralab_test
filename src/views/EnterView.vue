<template>
  <div class="wrapper">
    <div class="title">
      Ready for a great User experience?
    </div>
    <div class="subtitle">
      Bring your media to the next level!
    </div>
    <div class="button-group">
      <div
        class="button-group__signup"
        @click="$router.push('/sign_up')"
      >
        Sign Up
      </div>
      <img
        class="button-group__next"
        src="@/assets/arrow-right.svg"
        alt="&#10140;"
      />
    </div>
    <img
      class="enter-img"
      src="@/assets/woman_enter.svg"
      alt=""
    />
  </div>
</template>

<style lang="sass" scoped>
.wrapper
  justify-content: space-between

.subtitle
  text-align: center
  padding: 0.5em

.button-group
  display: flex
  flex-flow: row nowrap
  justify-content: center
  padding: 0.5em

  & > *:not(& > :last-child)
    margin-right: 2em

  &__signup
    +ellipse
    +shadow
    background-color: white

  &__next
    +round
    +shadow
    background-color: #1E1A3E
    color: white
    padding: 12px
</style>
