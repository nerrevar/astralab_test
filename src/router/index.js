import Vue from 'vue'
import VueRouter from 'vue-router'

import EnterView from '@/views/EnterView.vue'
import SignUpView from '@/views/SignUpView.vue'
import SignInView from '@/views/SignInView.vue'

Vue.use(VueRouter)

const routes = [
  {
    path: '/',
    name: 'EnterView',
    component: EnterView,
  },
  {
    path: '/sign_up',
    name: 'SignUpView',
    component: SignUpView,
  },
  {
    path: '/sign_in',
    name: 'SignInView',
    component: SignInView,
  },
  {
    path: '*',
    redirect: '/',
  }
]

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes,
})

export default router
