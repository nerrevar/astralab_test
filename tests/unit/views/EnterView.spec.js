import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'

import router from '@/router'

import EnterView from '@/views/EnterView.vue'

const localVue = createLocalVue()

localVue.use(VueRouter)

describe('EnterView.vue', () => {
  const vm = shallowMount(
    EnterView,
    {
      localVue,
      router,
    }
  )

  it('renders', () => {
    expect(vm.element).toMatchSnapshot()
  })
})
