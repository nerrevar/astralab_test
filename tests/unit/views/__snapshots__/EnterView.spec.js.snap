// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnterView.vue renders 1`] = `
<div
  class="wrapper"
>
  <div
    class="title"
  >
    
    Ready for a great User experience?
  
  </div>
   
  <div
    class="subtitle"
  >
    
    Bring your media to the next level!
  
  </div>
   
  <div
    class="button-group"
  >
    <div
      class="button-group__signup"
    >
      
      Sign Up
    
    </div>
     
    <img
      alt="&#10140;"
      class="button-group__next"
      src="@/assets/arrow-right.svg"
    />
  </div>
   
  <img
    alt=""
    class="enter-img"
    src="@/assets/woman_enter.svg"
  />
</div>
`;
